soh/src/overlays/actors/ovl_Obj_Lightswitch/z_obj_lightswitch.c

```diff
@@ -211,6 +247,26 @@ void ObjLightswitch_Destroy(Actor* thisx, GlobalContext* globalCtx2) {
     GlobalContext* globalCtx = globalCtx2;
     ObjLightswitch* this = (ObjLightswitch*)thisx;
 
+    // Unset the switch flag on room exit to prevent the rock in the wall from 
+    // vanishing on its own after activating the sun switch by Light Arrow
+    // Also prevents the cobra mirror from rotating to face the sun on its own
+    // Makes sun switches temporary when activated by Light Arrows (will turn off on room exit)
+    if (activatedByLightArrow && sunLightArrows) {
```

Comment:
does `activatedByLightArrow` ever get set to `true` when sunlight arrows aren't enabled? it seems that shouldn't ever happen, so this check can probably just be
```c
if (activatedByLightArrow) {
```

Reply:
Another good point.  I will correct this.  Probably me being overly cautious again :P

Reply:
Fixed! :)